### ISSUE TITLE
SA1000 and new expression

### ISSUE THREAD

**Issue by am11** [dotnet/roslyn#49728]
**Version Used**
3.8.0-5.20519.18 (4c195c3a)

#### Scenario
`dotnet format` (latest `v5.0.147901`) makes the following change to the code using C# 9 `new expression`:

```diff
class C
{
    static void M()
    {
-        C x = new ();
+        C x = new();
    }
}
```

This makes sense and desirable, when we are going from `var x = new C()` to `C x = new()` (saving three characters).

The CodeAnalysis.ruleset file has `SA1000` set to warning, and build is done with `/warnaserror` compiler flag.

#### Actual
`SA1000` gets violated and fails the build as:

```
C.cs(5,15): error SA1000: The keyword 'new' should be followed by a space.
```

#### Desired
`SA1000` is not violated.

#### Additional info
Around this place: https://github.com/dotnet/roslyn/blob/3bc4ff8f194ae055714a27b3aec4d1baa1ae8a9a/src/Workspaces/SharedUtilitiesAndExtensions/Compiler/CSharp/Formatting/Rules/TokenBasedFormattingRule.cs#L314
the rule is adjusted so the `new expressions` are treated as `new constraints`, which are also expected to have no space after the keyword.

**Comment by Youssef1313** [comment]
SA1000 isn't implemented in this repository. So this shouldn't be an issue with roslyn.

I'm looking now into its implementation.

**Comment by Youssef1313** [comment]
This was already fixed in https://github.com/DotNetAnalyzers/StyleCopAnalyzers/pull/3187

**Comment by am11** [comment]
Great, thanks for checking. It is indeed fixed in 1.2.0-beta.261 (was using 205 before).

I saw this issue from 2018 https://github.com/dotnet/roslyn/issues/28094, which was fixed in this repo. Has it moved since?

**Comment by Youssef1313** [comment]
@am11 Nope, the other issue is very different.

For the other issue, SA1000 is behaving correctly, and the formatter was behaving in a wrong way.
But for this issue, the formatter is behaving correctly, and the issue is with SA1000.
